src/libraries/System.IO.FileSystem/tests/Directory/Copy.cs

```diff
@@ -0,0 +1,216 @@
+            if (Directory.Exists(destDirName))
+            {
+                Directory.Delete(destDirName, true);
+            }
+            Directory.CreateDirectory(destDirName);
+
+            // Create file which does not exist in target folder
+            string firstPath = Path.Combine(sourceDirName, "first.txt");
+            using (StreamWriter firstStreamSource = File.CreateText(firstPath))
+            {
+                firstStreamSource.WriteLine("Test");
+            }
+
+            // Create file which does exist in target folder
+            string secondPathSource = Path.Combine(sourceDirName, "test.txt");
+            using (StreamWriter secondStreamSource = File.CreateText(secondPathSource))
+            {
+                secondStreamSource.WriteLine("Test");
+            }
+
+            string secondPathDest = Path.Combine(destDirName, "test.txt");
+            using (StreamWriter secondStreamDest = File.CreateText(secondPathDest))
+            {
+                secondStreamDest.WriteLine("Test");
+            }
+
+            // Do file system operation
+            Directory.Copy(sourceDirName, destDirName, false, false, cancellationToken);
+
+            // Validate cancellation
+            Assert.False(cancellationToken.IsCancellationRequested);
+        }
+
+        [Fact]
+        public void Copy_SkipExistingFiles()
+        {
+            const string sourceDirName = $"Test_source_{nameof(Copy_SkipExistingFiles)}";
+            const string destDirName = $"Test_target_{nameof(Copy_SkipExistingFiles)}";
+            CancellationToken cancellationToken = new();
+
+            // Prerequisites
+            // Create source folder
+            if (Directory.Exists(sourceDirName))
+            {
+                Directory.Delete(sourceDirName, true);
+            }
+            Directory.CreateDirectory(sourceDirName);
+
+            // Create destination folder
+            if (Directory.Exists(destDirName))
+            {
+                Directory.Delete(destDirName, true);
+            }
+            Directory.CreateDirectory(destDirName);
+
+            // Create file which does not exist in target folder
+            string firstPath = Path.Combine(sourceDirName, "first.txt");
+            using (StreamWriter firstStreamSource = File.CreateText(firstPath))
+            {
+                firstStreamSource.WriteLine("Test");
+            }
+
+            // Create file which does exist in target folder
+            string secondPathSource = Path.Combine(sourceDirName, "test.txt");
+            using (StreamWriter secondStreamSource = File.CreateText(secondPathSource))
+            {
+                secondStreamSource.WriteLine("Test");
+            }
+
+            string secondPathDest = Path.Combine(destDirName, "test.txt");
+            using (StreamWriter secondStreamDest = File.CreateText(secondPathDest))
+            {
+                secondStreamDest.WriteLine("Test");
+            }
```

Comment:
this is the exact same setup as the previous test, you should factor it out into a `Setup_FileAlreadyExistsInTarget` method.